src/App.tsx

```diff
@@ -1,18 +1,51 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './Pgination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState<number>(3);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const lastIndex = currentPage * perPage;
+  const firstIndex = lastIndex - perPage;
+
+  const itemsInPages = items.slice(firstIndex, lastIndex);
+  const pages = Math.ceil(items.length / perPage);
+  const numbersPages = Array.from({ length: pages }, (_, index) => index + 1);
+
+  const totalItems = items.length;
+  const displayedItems = lastIndex > totalItems ? totalItems : lastIndex;
+
+  const handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setCurrentPage(1);
+    setPerPage(+value);
+  };
+
+  const prePage = () => {
+    if (currentPage >= 1) {
```

Comment:
The condition should be 'currentPage > 1' instead of 'currentPage >= 1' to prevent going to a non-existing page.